destination.go

```diff
@@ -68,20 +74,39 @@ func (d *Destination) Configure(ctx context.Context, cfg map[string]string) erro
 	if err != nil {
 		return fmt.Errorf("invalid config: %w", err)
 	}
+	if !d.config.TLSDisable {
+		d.clientCert, d.caCertPool, err = d.config.ParseMTLSFiles()
+		if err != nil {
+			return fmt.Errorf("invalid mTLS config: %w", err)
+		}
+	}
 	return nil
 }
 
 func (d *Destination) Open(ctx context.Context) error {
 	dialOptions := []grpc.DialOption{
 		grpc.WithContextDialer(d.dialer),
-		grpc.WithInsecure(), //nolint:staticcheck // todo: will use mTLS with connection
+		grpc.WithTransportCredentials(insecure.NewCredentials()), // todo: will use mTLS with connection
 		grpc.WithBlock(),
 	}
 	if d.config.RateLimit > 0 {
 		dialOptions = append(dialOptions,
 			bwgrpc.WithBandwidthLimitedContextDialer(bwlimit.Byte(d.config.RateLimit), bwlimit.Byte(d.config.RateLimit), d.dialer))
 	}
-	conn, err := grpc.DialContext(ctx,
+	if !d.config.TLSDisable {
+		// create TLS credentials with mTLS configuration
+		creds := credentials.NewTLS(&tls.Config{
+			Certificates: []tls.Certificate{d.clientCert},
+			RootCAs:      d.caCertPool,
+			MinVersion:   tls.VersionTLS12,
```

Comment:
Is there a reason we don't set the minimum version to 1.3?

Reply:
We are in charge of both sides, client and server, I see no harm in setting this to 1.3.